src/bidiMapper/domains/network/NetworkManager.ts

```diff
@@ -43,12 +43,19 @@ export class NetworkManager {
     return this.#cdpTarget;
   }
 
-  #getOrCreateNetworkRequest(id: Network.Request): NetworkRequest {
+  /**
+   * Gets the network request with the given ID, if any.
+   * Otherwise, creates a new network request with the given ID and cdp target.
+   */
+  #getOrCreateNetworkRequest(
+    id: Network.Request,
+    cdpTarget: CdpTarget
```

Comment:
why do we need to pass the target around everywhere? is it just an id or the entire target?

Reply:
Alternatively we could pass the session ID, and retrieve the target from it.

Here's the issue: **We need to have the target available in the NetworkProcessor, so that it can issue commands to CDP. **

I was discussing it with @sadym-chromium yesterday, and the only effective way we found to do so is to get the target from the NetworkManager. And even then, we store the target in the NetworkRequest, not in the NetworkManager. Whenever there are some intercept commands, it then uses the request to retrieve the target.

If you can think of some other (better) way to inject the CdpTarget in the NetworkProcessor, I'd be happy to make changes to reflect that. It's just that, the way things are currently set up:

- NetworkManager is created from CdpTarget
- NetworkProcessor is created from CommandProcessor (which is created from BidiServer)

There's currently no straightforward way to inject the target into the processor. (Unless we use the NetworkStorage for that, but it would violate its contract.)

Reply:
I am not sure I have a good solution but the current versions looks somewhat confusing to me.

Reply:
the current version looks much better to me.